src/storage/query/QueryBaseProcessor.inl

```diff
@@ -409,6 +409,47 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
     }
     return ret;
 }
+template<typename REQ, typename RESP>
+kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
+                            PartitionID partId,
+                            VertexID vId,
+                            std::vector<cpp2::TagData> &tds) {
+    auto prefix = NebulaKeyUtils::vertexPrefix(partId, vId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = this->kvstore_->prefix(spaceId_, partId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return ret;
+    }
+
+    bool missedKey = true;
+    for (; iter && iter->valid(); iter->next()) {
+        if (!NebulaKeyUtils::isVertex(iter->key())) {
+            continue;
+        }
+        missedKey = false;
+        auto val = iter->val().toString();
+        auto ver = RowReader::getSchemaVer(val);
+        if (ver < 0) {
+            LOG(ERROR) << "Found schema version negative " << ver;
+            continue;
+        }
+        auto tagId = NebulaKeyUtils::getTagId(iter->key());
```

Comment:
function `key()` and `val()` returns the new string, and here call `key()` and `val()` multi times, you can define local var to save them to reduce string copy.

Reply:
We actually need to copy them.